examples/language-sdk-instrumentation/rust/README.md

```diff
@@ -1,2 +1,148 @@
+let car = warp::path("car").map(move || {
+    add("vehicle".to_string(), "car".to_string());
+    order_car(3);
+    remove("vehicle".to_string(), "car".to_string());
+    "Car ordered"
+});
+```
+
+What this block does, is:
+
+1. Add the label `vehicle=car`
+2. Execute the `order_car` function
+3. Remove the label `vehicle=car`
+
+## Resulting flame graph / performance results from the example
+
+### Running the example
+
+To run the example run the following commands:
+
+```
+# Pull latest pyroscope and grafana images:
+docker pull grafana/pyroscope:latest
+docker pull grafana/grafana:latest
+
+# Run the example project:
+docker-compose up --build
+
+# Reset the database (if needed):
+# docker-compose down
+```
+
+What this example will do is run all the code mentioned above and also send some mock-load to the 3 servers as well as
+their respective 3 endpoints. If you select our application: `rust-ride-sharing-app` from the dropdown, you should see a
+flame graph that looks like this (below). After we give 20-30 seconds for the flame graph to update and then click the
+refresh button we see our 3 functions at the bottom of the flame graph taking CPU resources _proportional to the size_
+of their respective `search_radius` parameters.
+
+[//]: # (http://localhost:3000/a/grafana-pyroscope-app/profiles-explorer?searchText=&panelType=time-series&layout=grid&hideNoData=off&explorationType=flame-graph&var-serviceName=rust-ride-sharing-app&var-profileMetricId=process_cpu:cpu:nanoseconds:cpu:nanoseconds&var-dataSource=local-pyroscope&var-groupBy=all&var-filters=&maxNodes=16384&from=now-5m&to=now&var-filtersBaseline=&var-filtersComparison=)
+
+## Where's the performance bottleneck?
+
+![Image](https://github.com/user-attachments/assets/d4b0f85d-cc8d-4058-b019-1c5198849676)
+
+The first step when analyzing a profile outputted from your application, is to take note of the _largest node_ which is
+where your application is spending the most resources. In this case, it happens to be the `order_car` function.
+
+The benefit of using the Pyroscope package, is that now that we can investigate further as to _why_ the `order_car()`
+function is problematic. Tagging both `region` and `vehicle` allows us to test two good hypotheses:
+
+- Something is wrong with the `/car` endpoint code
+- Something is wrong with one of our regions
+
+To analyze this we can select one or more tags on the "Labels" page:
+
+![Image](https://github.com/user-attachments/assets/3e5cb3ac-609e-493a-ae4d-248de150a33b)
+
+## Narrowing in on the Issue Using Tags
+
+Knowing there is an issue with the `order_car` function we automatically select that tag. Then, after inspecting
+multiple `region` tags, it becomes clear by looking at the timeline that there is an issue with the `eu-north` region,
+where it alternates between high-cpu times and low-cpu times.
+
+We can also see that the `mutex_lock()` function is consuming almost 70% of CPU resources during this time period.
```

Comment:
```suggestion
Since you know there is an issue with the `order_car` function,  select that tag. After inspecting
multiple `region` tags, the timeline shows that there is an issue with the `eu-north` region,
where it alternates between high-cpu times and low-cpu times.

Note that the `mutex_lock()` function is consuming almost 70% of CPU resources during this time period.
```